src/twisted/internet/cfreactor.py

```diff
@@ -132,7 +132,7 @@ def __init__(self, runLoop=None, runner=None):
         self._runner = runner
 
         if runLoop is None:
-            runLoop = CFRunLoopGetMain()
+            runLoop = CFRunLoopGetCurrent()
```

Comment:
`CFRunLoopGetMain` gets "the" main loop.  I suppose this is "the main thread's" event loop?
`CFRunLoopGetCurrent` gets "the current thread's" main loop.

The latter seems more likely to be correct in general.

* https://developer.apple.com/documentation/corefoundation/1542428-cfrunloopgetcurrent 
* https://developer.apple.com/documentation/corefoundation/1542890-cfrunloopgetmain

Both doc pages say the API follows "The Get Rule".  Do the Python bindings deal with that for us or are we supposed to be making `CFRetain` calls from Python (@glyph )?

Reply:
Correct on both counts. It's pyobjc's job, generally, to deal with retention rules.